### PR TITLE
WFCORE-5375 Update all projects with a dependency on deprecated Elytron modules to no longer rely on deprecated classes

### PR DESCRIPTION
**Pull request by parsharma**
Issue: https://issues.redhat.com/browse/WFCORE-5375

### CODE REVIEW

File: elytron/src/test/java/org/wildfly/extension/elytron/LdapTestCase.java [hunk 1]
```diff
@@ -146,6 +147,7 @@ private static void loadResources() throws Exception {
     public static void startLdapService() throws Exception {
         loadResources();
         TestEnvironment.startLdapService();
+        Security.setProperty("keystore.PKCS12.keyProtectionAlgorithm", "PBEWithHmacSHA256AndAES_128");
```
Comment (fjuma) [inline elytron/src/test/java/org/wildfly/extension/elytron/LdapTestCase.java:150]: There's a PR that fixes `LdapTestCase` (https://github.com/wildfly/wildfly-core/pull/4577) so this change can be removed here.

File: core-feature-pack/common/pom.xml [hunk 2]
```diff
@@ -301,10 +301,6 @@
             <groupId>org.wildfly.security</groupId>
             <artifactId>wildfly-elytron-credential</artifactId>
         </dependency>
-        <dependency>
-            <groupId>org.wildfly.security</groupId>
-            <artifactId>wildfly-elytron-credential-source-deprecated</artifactId>
-        </dependency>
```
Comment (fjuma) [inline core-feature-pack/common/pom.xml:307]: There are also a few more dependencies on other deprecated Elytron modules that should be removed here as well:
* `wildfly-elytron-auth-server-deprecated`
* `wildfly-elytron-http-deprecated`
* `wildfly-elytron-sasl-deprecated`
* `wildfly-elytron-x500-deprecated`

Comment (fjuma) [reply core-feature-pack/common/pom.xml:307]: The dependency on `wildfly-elytron-x500-deprecated` in `elytron/pom.xml` should also be removed.

File: pom.xml [hunk 3]
```diff
@@ -1789,11 +1789,6 @@
                 <artifactId>wildfly-elytron-credential</artifactId>
                 <version>${version.org.wildfly.security.elytron}</version>
             </dependency>
-            <dependency>
-                <groupId>org.wildfly.security</groupId>
-                <artifactId>wildfly-elytron-credential-source-deprecated</artifactId>
-                <version>${version.org.wildfly.security.elytron}</version>
-            </dependency>
```
Comment (fjuma) [inline pom.xml:1796]: There are also a few more dependencies on other deprecated Elytron modules that should be removed here as well:
* `wildfly-elytron-auth-server-deprecated`
* `wildfly-elytron-http-deprecated`
* `wildfly-elytron-sasl-deprecated`
* `wildfly-elytron-x500-deprecated`

File: core-feature-pack/common/pom.xml (hunk 2 above)
Comment (fjuma) [reply core-feature-pack/common/pom.xml:307]: The `wildfly-elytron-auth-server-deprecated` dependency should also be removed from `protocol/pom.xml` and `remoting/subsystem/pom.xml`.

**Review by fjuma — CHANGES_REQUESTED** [review]
Thanks very much for working on this, @parsharma! I've added some comments about dependencies on other deprecated Elytron modules that should be removed as well.

File: elytron/pom.xml [hunk 4]
```diff
@@ -189,6 +189,11 @@
             <artifactId>wildfly-elytron-sasl</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.wildfly.security</groupId>
+            <artifactId>wildfly-elytron-sasl-auth-util</artifactId>
+        </dependency>
```
Comment (fjuma) [inline elytron/pom.xml:195]: Just to check, is anything making use of this? I don't see any imports adjusted related to this one.

**Review by fjuma — COMMENTED** [review]
@parsharma Thanks very much for working on this and for creating the WFCORE issues for the follow-up work! Just have one minor comment.

File: elytron/pom.xml (hunk 4 above)
Comment (parsharma) [reply elytron/pom.xml:195]: Yes, I have checked no imports are dependent on this dependency. As I remember some test case was failing in wildly but I have re-checked everything is fine. I will push the updated one again.

**Pull request merged**